### PR TITLE
Correctly pad oidc tokens

### LINKED ISSUE #65: OIDC auth uses incorrect base64 decoding
We had a customer report the following error with this client:

```
Traceback (most recent call last): 
File ".../k8s_client.py", line 6, in <module> 
config.load_kube_config() 
File ".../venv/lib/python3.6/site-packages/kubernetes/config/kube_config.py", line 473, in load_kube_config 
loader.load_and_set(config) 
File ".../venv/lib/python3.6/site-packages/kubernetes/config/kube_config.py", line 354, in load_and_set 
self._load_authentication() 
File ".../venv/lib/python3.6/site-packages/kubernetes/config/kube_config.py", line 185, in _load_authentication 
if self._load_oid_token(): 
File ".../venv/lib/python3.6/site-packages/kubernetes/config/kube_config.py", line 236, in _load_oid_token 
base64.b64decode(parts[1]).decode('utf-8') 
File ".../python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/base64.py", line 87, in b64decode 
return binascii.a2b_base64(s) 
binascii.Error: Incorrect padding
```

Which appears to originate from here: https://github.com/kubernetes-client/python-base/blob/789de6a60dea9442ee47601a788d92c653ea69e9/config/kube_config.py#L236

```
jwt_attributes = json.loads(
    base64.b64decode(parts[1]).decode('utf-8')
)
```

JWTs aren't encoded using standard base64 encoding, they use URL encoding without the final padding

```
   Base64url Encoding
      Base64 encoding using the URL- and filename-safe character set
      defined in Section 5 of RFC 4648 [RFC4648], with all trailing '='
      characters omitted (as permitted by Section 3.2) and without the
      inclusion of any line breaks, whitespace, or other additional
      characters.  Note that the base64url encoding of the empty octet
      sequence is the empty string.  (See Appendix C for notes on
      implementing base64url encoding without padding.)
```

https://tools.ietf.org/html/rfc7515#section-2

So "hello world" should become `aGVsbG8gd29ybGQ`, not `aGVsbG8gd29ybGQ=` https://play.golang.org/p/vFrVzr9uyAQ

Python's default base64 library doesn't handle this encoding and spits out the same exception our customer's seeing:

```
$ python3 -c 'import base64; base64.b64decode("aGVsbG8gd29ybGQ")'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib64/python3.6/base64.py", line 87, in b64decode
    return binascii.a2b_base64(s)
binascii.Error: Incorrect padding
```

### PR DESCRIPTION
**Pull request by hanikesn**
According to the JWT spec base64 padding characters are stripped.
Fixes #65

Tested only with python 3.

### CODE REVIEW

**Comment by bcatubig** [comment]
Hey @hanikesn ,

Would you mind updating your branch? Would love to see this get merged in.

**Comment by richerve** [comment]
@hanikesn are you still interested in updating this PR?

File: config/kube_config_test.py [hunk 1]
```diff
@@ -87,11 +87,11 @@ def _raise_exception(st):
 
 TEST_OIDC_TOKEN = "test-oidc-token"
 TEST_OIDC_INFO = "{\"name\": \"test\"}"
-TEST_OIDC_BASE = _base64(TEST_OIDC_TOKEN) + "." + _base64(TEST_OIDC_INFO)
+TEST_OIDC_BASE = _base64(TEST_OIDC_TOKEN).strip('=') + "." + _base64(TEST_OIDC_INFO).strip('=')
```
Comment (markuszoeller) [inline config/kube_config_test.py]: Looks like the length of the changed lines in this file is what makes the travis job complain. I get this output when I test the styleguide locally:
```console
$ pycodestyle config/kube_config*
config/kube_config_test.py:90:80: E501 line too long (95 > 79 characters)
config/kube_config_test.py:94:80: E501 line too long (98 > 79 characters)
```

**Comment by hanikesn** [comment]
I will take another look this weekend.

**Comment by iMartyn** [comment]
Eagerly waiting for this fix to be merged so that kube-ops-view can pull it in and then we can have our monitoring working for OIDC clusters.

**Comment by kmai** [comment]
is this stale? OIDC is still broken with the k8s python client..

**Comment by danigrmartinez** [comment]
@hanikesn this fix works for me. I would love to see this merged
Thanks for the fix.

**Comment by micw523** [comment]
Could you use autopep8 on aggressive level 2 to prepare your code? That should let you pass the Travis CI. @hanikesn

File: config/kube_config_test.py (hunk 1 above)
Comment (micw523) [inline config/kube_config_test.py]:
```suggestion
TEST_OIDC_BASE = _base64(TEST_OIDC_TOKEN).strip(
    '=') + "." + _base64(TEST_OIDC_INFO).strip('=')
```

File: config/kube_config_test.py [hunk 2]
```diff
@@ -87,11 +87,11 @@ def _raise_exception(st):
...
 TEST_OIDC_LOGIN = TEST_OIDC_BASE + "." + TEST_CLIENT_CERT_BASE64
 TEST_OIDC_TOKEN = "Bearer %s" % TEST_OIDC_LOGIN
 TEST_OIDC_EXP = "{\"name\": \"test\",\"exp\": 536457600}"
-TEST_OIDC_EXP_BASE = _base64(TEST_OIDC_TOKEN) + "." + _base64(TEST_OIDC_EXP)
+TEST_OIDC_EXP_BASE = _base64(TEST_OIDC_TOKEN).strip('=') + "." + _base64(TEST_OIDC_EXP).strip('=')
```
Comment (micw523) [inline config/kube_config_test.py]:
```suggestion
TEST_OIDC_EXP_BASE = _base64(TEST_OIDC_TOKEN).strip(
    '=') + "." + _base64(TEST_OIDC_EXP).strip('=')
```

**Comment by wwade** [comment]
@hanikesn @micw523 has this change been abandoned?  I can create a new PR if needed and help it cross the finish line.

**Comment by micw523** [comment]
@wwade This one was abandoned I think. The other one is still open.

**Comment by rocktavious** [comment]
@wwade @micw523 i'd really like to track the status of this as its blocking me from writing a tool with python against our OIDC backed k8s clusters.  Not sure if there is a different PR to watch.

**Comment by ralphje** [comment]
> @wwade @micw523 i'd really like to track the status of this as its blocking me from writing a tool with python against our OIDC backed k8s clusters. Not sure if there is a different PR to watch.

I guess you could also watch #79

**Comment by roycaihw** [comment]
Closing in favor of https://github.com/kubernetes-client/python-base/pull/79. Thanks for the fix!

**Pull request closed without merging**